src/components/Pagination.tsx

```diff
@@ -0,0 +1,33 @@
+import { Button } from 'flowbite-react';
+import React from 'react';
+
+interface PaginationProps {
+  currentPage: number;
+  totalPages: number;
+  onNextPage: () => void;
+  onPrevPage: () => void;
+  onGoToBeginning: () => void;
+  onGoToLast: () => void;
+}
+
+const Pagination: React.FC<PaginationProps> = ({ currentPage, totalPages, onNextPage, onPrevPage, onGoToBeginning, onGoToLast }) => {
+  return (
+    <div className="flex items-center justify-center my-2 gap-4">
+      <Button onClick={onGoToBeginning} disabled={currentPage === 1} className='bg-black enabled:hover:bg-gray-600 dark:bg-white dark:enabled:hover:bg-gray-600 dark:enabled:hover:text-white dark:text-black'>
+        First
+      </Button>
+      <Button onClick={onPrevPage} disabled={currentPage === 1} className='bg-black enabled:hover:bg-gray-600 dark:bg-white dark:enabled:hover:bg-gray-600 dark:enabled:hover:text-white dark:text-black'>
+        Previous
+      </Button>
+      <span>{`Page ${currentPage} of ${totalPages}`}</span>
```

Comment:
This should also match the theme - can't really see it on dark mode
![image](https://github.com/ProvinceWatch/open-watch/assets/49263118/d00fa899-5f7a-45b8-bddc-1e77edc4c262)